aodncore/util/misc.py

```diff
@@ -95,6 +96,24 @@ def format_exception(exception):
     return "{cls}: {message}".format(cls=exception.__class__.__name__, message=exception)
 
 
+def get_fields_from_filename(filepath, pattern):
+    """Function to retrieve parts of a filename given a compiled pattern (re.compile(str))
+    :return: dictionary of fields matching a given pattern
+    """
+    retype = type(re.compile(''))
+    if not isinstance(pattern, retype):
+        try:
+            pattern = re.compile(pattern)
+        except Exception as err:
+            raise TypeError("pattern: \"{pattern}\" could not be compiled as a \"{retype}\" instance".format(
+                pattern=pattern,
+                retype=retype))
+
+    file_basename = os.path.basename(filepath)
```

Comment:
This is definitely a potential point of confusion... if you have a path and a regex it's not intuitive that inside the function, it would silently match against the basename.

Maybe we should add a 'basename' parameter which defaults to False, so that if you want to match against only the basename, it's obvious to the caller.

Reply:
I guess this is why the name of the function is
get_fields_from_filename

Doing the os.path.basename was to force this in case a full path is the input of the function